src/main/java/me/drendov/HouseKeeping/HouseKeeping.java

```diff
@@ -191,4 +194,37 @@ synchronized public String getMessage(Messages messageID, String... args) {
     }
 
 
+    private void checkDependency(String name) {
+        if (this.getServer().getPluginManager().getPlugin(name) == null) {
+            logger.info("Missing dependency '" + name + "'. Cannot continue.");
+            getPluginLoader().disablePlugin(this);
+            return;
+        }
+    }
+
+
+    private void checkAbsence() {
+        Essentials ess = (Essentials)Bukkit.getPluginManager().getPlugin("Essentials");
+        File csv = new File(ess.getDataFolder(), "usermap.csv");
+        UserMap userMap = ess.getUserMap();
```

Comment:
I don't think that we can count on this file.
Have you had a look at getOfflinePlayers() or getOnlinePlayers()?

Reply:
`Bukkit.getOfflinePlayers()` does not report any players listed in `plugins/Essentials/playerdata`. Essentials does not seem to provide any cleaner way to list all players. `UserMap` should provide a method to get all uniuqe players, but it seems to return empty set all the time during testing.